File: levelledmobs-plugin-bukkit/src/main/java/me/lokka30/levelledmobs/bukkit/event/action/ActionPreParseEvent.java
package me.lokka30.levelledmobs.bukkit.event.action;

import java.util.Objects;
import me.lokka30.levelledmobs.bukkit.event.CancellableEvent;
import me.lokka30.levelledmobs.bukkit.logic.Action;
import org.bukkit.event.HandlerList;
import org.jetbrains.annotations.NotNull;

public class ActionPreParseEvent extends CancellableEvent implements ActionEvent {

    /* vars */

    private static final HandlerList HANDLERS = new HandlerList();
    private final Action action;

    /* constructors */

    public ActionPreParseEvent(final @NotNull Action action) {
        this.action = Objects.requireNonNull(action, "action");
    }

    /* getters and setters */

    @Override
    @NotNull
    public Action getAction() { return action; }

    @NotNull
    @Override
    public HandlerList getHandlers() {
        return HANDLERS;
    }
}

File: levelledmobs-plugin-bukkit/src/main/kotlin/io/github/arcaneplugins/levelledmobs/plugin/bukkit/event/rule/ActionPreParseEvent.kt
package io.github.arcaneplugins.levelledmobs.plugin.bukkit.event.rule

import io.github.arcaneplugins.levelledmobs.plugin.bukkit.rule.component.action.Action
import org.bukkit.event.Cancellable
import org.bukkit.event.Event
import org.bukkit.event.HandlerList

//todo doc
class ActionPreParseEvent(
    val action: Action,
) : Event(), Cancellable {

    companion object {
        val HANDLERS: HandlerList = HandlerList()
    }

    private var cancelled: Boolean = false

    override fun getHandlers(): HandlerList {
        return HANDLERS
    }

    override fun isCancelled(): Boolean {
        return cancelled
    }

    override fun setCancelled(cancelled: Boolean) {
        this.cancelled = cancelled
    }

}